### ISSUE TITLE
Unhandled promise rejection with stack size overflow in node-vault

### ISSUE THREAD

**Issue by olegdulin** [kr1sp1n/node-vault#80]
The call to validate happens as part of my application health check monitoring. I am going to temporarily take it out until a more long term solution is found. If there is a workaround please let me know.

We are using version 0.6.1 and I am going to try updating the library first, in case this was fixed already.

It took me awhile to get to the bottom of why my process was crashing, but here is the stack trace (I spared you the middle 20000 lines :) ):

```
unhandledRejection:Maximum call stack size exceeded stack:RangeError: Maximum call stack size exceeded
    at isPlainObject (/app/node_modules/extend/index.js:14:43)
    at extend (/app/node_modules/extend/index.js:64:27)
    at extend (/app/node_modules/extend/index.js:73:22)
    at /app/node_modules/request/index.js:89:5
    at /app/node_modules/request/index.js:101:12
    at /app/node_modules/request/index.js:101:12
  ...
    at /app/node_modules/request/index.js:101:12
    at /app/node_modules/request/index.js:101:12
    at /app/node_modules/request/index.js:101:12
    at /app/node_modules/request/index.js:101:12
    at /app/node_modules/request/index.js:101:12
    at /app/node_modules/request/index.js:101:12
    at /app/node_modules/request/index.js:101:12
    at Object.client.request (/app/node_modules/node-vault/src/index.js:79:12)
    at validate.then.then.then (/app/node_modules/node-vault/src/index.js:163:41)
    at <anonymous>
    at process._tickDomainCallback (internal/process/next_tick.js:228:7)```

**Comment by DaniGuardiola** [comment]
@olegdulin Could you please test [this branch](https://github.com/daniguardiola/node-vault/tree/refactor)?

[How to](https://medium.com/@jonchurch/use-github-branch-as-dependency-in-package-json-5eb609c81f1a) :)

**Comment by DaniGuardiola** [comment]
Marked as wontfix, the codebase has changed completely in https://github.com/kr1sp1n/node-vault/pull/78 and this issue should arise anymore once `v1.0.0` is released.

**Comment by johnmaia** [comment]
Hi there,
I'm currently using the version `0.9.8` and I'm still getting a similar error:

```
Object.client.request (/home/node/app/node_modules/node-vault/src/index.js:91:12)\n
at Object.client.read (/home/node/app/node_modules/node-vault/src/index.js:116:19)\n  
at Object.read (/quux/vault-client.js:1:3359)\n 
at FoobarManager.getFoobiz (/home/node/app/dist/foobar-manager.js:1:4807)\n 
at _0x351791.(anonymous function) 
{"name":"foobar-manager","hostname":"foobar-manager-ff546cd5b-ts5dg","pid":1,
"component":"clients","level":50,"subcomponent":"vault",
"msg":"Unable to read value from the path qux/foobiz:  RangeError: Maximum call stack size exceeded\n
at extend (/home/node/app/node_modules/extend/index.js:65:33)\n    
at initParams (/home/node/app/node_modules/request/index.js:35:5)\n    
at /home/node/app/node_modules/request/index.js:85:18\n    
at /home/node/app/node_modules/request/index.js:100:12\n    
at /home/node/app/node_modules/request/index.js:100:12\n    
at /home/node/app/node_modules/request/index.js:100:12\n    
at /home/node/app/node_modules/request/index.js:100:12\n    
at /home/node/app/node_modules/request/index.js:100:12\n    
at /home/node/app/node_modules/request/index.js:100:12\n    
at /home/node/app/node_modules/request/index.js:100:12\n    
at /home/node/app/node_modules/request/index.js:100:12\n    
at /home/node/app/node_modules/request/index.js:100:12\n    
at /home/node/app/node_modules/request/index.js:100:12\n    
at /home/node/app/node_modules/request/index.js:100:12\n    
at /home/node/app/node_modules/request/index.js:100:12\n    
at /home/node/app/node_modules/request/index.js:100:12\n    
at /home/node/app/node_modules/request/index.js:100:12\n
...
```

**Comment by kr1sp1n** [comment]
@olegdulin @johnmaia Should be fixed now with v0.9.11. Please test again.

**Comment by alietors** [comment]
Hi,

It's possible that this error came back on v0.9.20?
